### PR TITLE
Include dedx collections in fastsim output

### PR DESCRIPTION
**Pull request by kpedro88**
#### PR description:

The `FastTrackDeDxProducer` is now included in the FastSim reconstruction sequence with two versions (strip tracker and pixel).

The sizes of the new output collections from a test config are (using `edmEventSize`):
```
Branch Name | Average Uncompressed Size (Bytes/Event) | Average Compressed Size (Bytes/Event) 
recoDeDxDataedmValueMap_dedxHarmonic2__RECO. 11165.7 1507.67
recoDeDxDataedmValueMap_dedxPixelHarmonic2__RECO. 11192.3 1399.33
```

Similar collections are already included in regular RECO/AOD output, so this should be an acceptable increase for FastSim RECO/AOD output.

This PR also fixes a condition in the producer that was modified in the backport PR #28235, but erroneously not modified in master.

#### PR validation:

Ran a private test config without errors and observed the desired collections in the output.

attn: @sbein @ssekmen @scarletnorberg

### CODE REVIEW

**Comment by kpedro88** [comment]
please test

File: RecoTracker/Configuration/python/RecoTracker_cff.py [hunk 1]
```diff
@@ -38,5 +38,6 @@
 trackingGlobalReco = cms.Sequence(trackingGlobalRecoTask)
 
 from Configuration.Eras.Modifier_fastSim_cff import fastSim
-_fastSim_trackingGlobalReco = cms.Sequence(trackExtrapolator)
+ckftracks_fast = cms.Sequence(doAlldEdXEstimators)
+_fastSim_trackingGlobalReco = cms.Sequence(ckftracks_fast*trackExtrapolator)
 fastSim.toReplaceWith(trackingGlobalReco,_fastSim_trackingGlobalReco)
```
Comment (makortel) [inline RecoTracker/Configuration/python/RecoTracker_cff.py]: How about the following?
```python
fastSim.toReplaceWith(trackingGlobalRecoTask, cms.Task(doAlldEdXEstimators, trackExtrapolator))
```

File: RecoTracker/DeDx/python/dedxEstimators_cff.py [hunk 2]
```diff
@@ -78,3 +92,9 @@
 
 doAlldEdXEstimatorsTask = cms.Task(dedxTruncated40 , dedxHarmonic2 , dedxPixelHarmonic2 , dedxPixelAndStripHarmonic2T085 , dedxHitInfo)
 doAlldEdXEstimators = cms.Sequence(doAlldEdXEstimatorsTask)
+
+doFastdEdXEstimatorsTask = cms.Task(dedxHarmonic2, dedxPixelHarmonic2)
+doFastdEdXEstimators = cms.Sequence(doFastdEdXEstimatorsTask)
+
+fastSim.toReplaceWith(doAlldEdXEstimatorsTask, doFastdEdXEstimatorsTask)
+fastSim.toReplaceWith(doAlldEdXEstimators, doFastdEdXEstimators)
```
Comment (makortel) [inline RecoTracker/DeDx/python/dedxEstimators_cff.py]: To me `doFastdEdXEstimators` Sequence looks redundant as the necessary replacement is done already with `doFastdEdXEstimatorsTask` (which could be further simplified to
```python
fastSim.toReplaceWith(doAlldEdXEstimatorsTask, cms.Task(dedxHarmonic2, dedxPixelHarmonic2))
```
)

File: RecoTracker/Configuration/python/RecoTracker_cff.py (hunk 1 above)
Comment (kpedro88) [reply RecoTracker/Configuration/python/RecoTracker_cff.py]: It actually needs to be `doAlldEdXEstimatorsTask` since sequences can't be added to tasks

**Comment by kpedro88** [comment]
please test

**Comment by sbein** [comment]
+1

**Comment by perrotta** [comment]
+1
- Changes in the reco configs only refer to fastsim customizations
- Jenkins tests pass and show no changes in reco outputs

**Comment by fabiocos** [comment]
+1

**Pull request merged**